rpcs3/Emu/Cell/lv2/sys_spu.cpp

```diff
@@ -648,9 +648,10 @@ error_code sys_spu_thread_group_join(ppu_thread& ppu, u32 id, vm::ptr<u32> cause
 			return CELL_EBUSY;
 		}
 
-		if (group->run_state == SPU_THREAD_GROUP_STATUS_INITIALIZED)
+		if (group->join_signal)
```

Comment:
I believe it can be implemented without adding join_signal member.
Instead, this condition could become the conjunction of:
1) Previous `run_state == SPU_THREAD_GROUP_STATUS_INITIALIZED`
2) join_state is set to a meaningful value (one of three cases, will need to modify its usage a bit).

SPU_TGJSF_ constants would need to be removed, they are obsolete. Original SPU constants for `cause` could be used directly in join_state (SYS_SPU_THREAD_GROUP_JOIN_).

Reply:
I think combining the two checks is a bit overkill, a flag in join_state should be enough.

Reply:
Why would it be an overkill? Flags are not necessarily easier to write or faster. And condition here is just one place (second condition can even continue to use stop_count), when a flag will need to be carefully used in several places.